pkg/build/generator/generator.go

```diff
@@ -585,10 +583,8 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from kapi
 	case "ImageStreamTag":
 		imageStreamTag, err := g.Client.GetImageStreamTag(kapi.WithNamespace(ctx, namespace), from.Name)
 		if err != nil {
-			glog.V(2).Infof("Error resolving ImageStreamTag reference %s in namespace %s: %v", from.Name, namespace, err)
-			if errors.IsNotFound(err) {
-				return "", err
-			}
+			err = fmt.Errorf("Error resolving ImageStreamTag reference %s in namespace %s: %v", from.Name, namespace, err)
```

Comment:
this needs to be wrapped into a NewNotFound error or some other appropriate StatusError so it comes back as a proper http status code error.  (see kubernetes/pkg/api/errors/errors.go)

same for the block above.